.github/workflows/changelog.yml

```diff
@@ -1,34 +1,34 @@
-name: Check Changelog.md
+name: Check PR for Changelog
+
 on:
   pull_request:
-    types: [opened, synchronize, reopened]
+    types: [opened, edited, reopened]
+
 jobs:
-  check-changelog:
-     runs-on: ubuntu-latest
-     permissions: write-all
-     steps:
-          - uses: actions/checkout@v4
-          - name: Fetch changed files
-            id: fetch-changed-files
-            run: |
-              response=$(gh api \
-              -H "Accept: application/vnd.github+json" \
-              -H "X-GitHub-Api-Version: 2022-11-28" \
-               repos/${owner}/${REPO}/pulls/${PULL_NUMBER}/files)
-              echo "$response" | jq '.' > response.json
-              
-          
-            env:
-              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-              owner: ${{ github.repository_owner }}
-              REPO: ${{ github.event.repository.name }}
-              PULL_NUMBER: ${{ github.event.pull_request.number }}
-          - name: Check if Changelog.md is changed
-            run: |
-              if [[ $(jq '.[].filename' response.json) == *"CHANGELOG.md"* ]]; then
-                echo "Changelog.md is changed"
-                exit 0
-              else
-                echo "Changelog.md is not changed"
-                exit 1
-              fi
+  check-description:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check PR Description
+        id: check_description
+        run: |
+          description=$(jq -r ".pull_request.body" "$GITHUB_EVENT_PATH")
+          if [[ "$description" == *"## Changelog"* ]]; then
```

Comment:
Not required, would just be nice-to-have. People sometimes open a PR and just leave it unfilled, like this

![Screenshot from 2024-02-22 19-46-41](https://github.com/learningequality/kolibri-design-system/assets/13509191/ae8953b5-3230-4fd5-8608-469e26d7454f)

If we could check that for example "Description" was changed from its default value, that would help from merging such items accidentally. Or we could do something else along this line if you have any better idea.

Reply:
I agree to that. Description is one field which we expect every contributor to fill out. So it makes sense to check if it has been updated or not to further verify the presence of changelog section. 
Also, while going through everything, I realized that I should have included a check to see if the PR is made by dependabot in the check-changelog workflow. If we don't have that, the action will keep failing for dependabot PRs. 
If you agree to the above changes, I can start working on them.

Reply:
Yes, that makes sense to me, thank you

Reply:
> while going through everything, I realized that I should have included a check to see if the PR is made by dependabot in the check-changelog workflow

We're okay with merging with checks failing as long as there's a good reason for so, so it wouldn't be anything blocking. Nice to have for sure, as the PR will be a bit more clear.

Reply:
Just a note that even when we can have more changelog items in the PR description as I've just mentioned below, it's absolutely okay to only check for the first "Description" being changed since we basically just want to detect that someone changed something.